### PR TITLE
ResourceLoader: Fixup resource changed feature

### PR DESCRIPTION
**Pull request by RandomShaper**
This is a complement to #96593.

### CODE REVIEW

**Comment by RandomShaper** [comment]
Cherry-pick for 4.3 included in #96606.

File: core/io/resource_loader.cpp [hunk 1]
```diff
@@ -818,6 +818,8 @@ Ref<Resource> ResourceLoader::_load_complete_inner(LoadToken &p_load_token, Erro
 		load_task_ptr = &load_task;
 	}
 
+	thread_load_mutex.unlock();
```
Comment (AThousandShips) [inline core/io/resource_loader.cpp:821]: Given the scope I'd use the lock, like so:
```suggestion
	p_thread_load_lock.temp_unlock();
```

File: core/io/resource_loader.cpp [hunk 2]
```diff
@@ -855,6 +857,8 @@ Ref<Resource> ResourceLoader::_load_complete_inner(LoadToken &p_load_token, Erro
 		}
 	}
 
+	thread_load_mutex.lock();
```
Comment (AThousandShips) [inline core/io/resource_loader.cpp:860]:
```suggestion
	p_thread_load_lock.temp_relock();
```

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**